services/arrow-flight/src/main/java/com/dremio/service/flight/impl/FlightWorkManager.java

```diff
@@ -253,25 +269,73 @@ public void runGetTables(FlightSql.CommandGetTables commandGetTables,
 
     final UserProtos.GetTablesResp getTablesResp = responseHandler.get();
 
-    try (VectorSchemaRoot vectorSchemaRoot = VectorSchemaRoot.create(FlightSqlProducer.Schemas.GET_TABLES_SCHEMA,
-      allocator)) {
+    final boolean includeSchema = commandGetTables.getIncludeSchema();
```

Comment:
Can you refactor to reduce the size of this method? Rough rule of thumb is try to keep to 100 lines per method.

Reply:
Nice, I've refactored